### PR TITLE
Monad types + utils in Treajure 

### PR DESCRIPTION
**Pull request by roman**

### CODE REVIEW

File: src/com/unbounce/treajure/either.clj [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  [val]
+  (if (monadic-value? val)
+    (monads.types/left? (run-either val))
+    (monads.types/left? val)))
+
+(defmacro try-either
+  "Performs a call that might fail and wraps its result on an Either.
+When the call throws an exception, the value is wrapped on a `left' value;
+  otherwise a `right' value"
+  [io-action]
```
Comment (artgon) [inline src/com/unbounce/treajure/either.clj]: A function on failure would be nice to have, as well.

File: src/com/unbounce/treajure/monad.clj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+(defmethod monadic-value? :default [_]
+  false)
+
+(defmacro do [& instr]
+  `(monads/mdo ~@instr))
+
+(def return monads/return)
+
+(defn bind-ignore
+  "Executes monadic sub-routine withhout binding it's return value"
```
Comment (artgon) [inline src/com/unbounce/treajure/monad.clj]: Would be good to expand this comment a bit with an example of why this is useful.

Comment (roman) [inline src/com/unbounce/treajure/monad.clj]: :+1:

File: src/com/unbounce/treajure/either.clj (hunk 1 above)
Comment (roman) [inline src/com/unbounce/treajure/either.clj]: :+1:

**Pull request merged**